### PR TITLE
[RNMobile] Add logging to Page Template Picker

### PR DESCRIPTION
**Pull request by chipsnyder**
**Fixes:** https://github.com/wordpress-mobile/gutenberg-mobile/issues/1883
**Related PRs:**
- `gutenberg-mobile iOS and JS` https://github.com/wordpress-mobile/gutenberg-mobile/pull/1936
- `WordPress-iOS` https://github.com/wordpress-mobile/WordPress-iOS/pull/13510
- `WordPress-iOS-Shared` https://github.com/wordpress-mobile/WordPress-iOS-Shared/pull/250
- `gutenberg-mobile Android` https://github.com/wordpress-mobile/gutenberg-mobile/pull/1945
- `WordPress-Android` https://github.com/wordpress-mobile/WordPress-Android/pull/11362

## Description

This adds the `template` property to the logging to the `editor_session_end` event if a user created a page and selected one of the starter page templates. 

### Create a new page using a template 
1) Open the WordPress iOS app 
2) Navigate to pages and create a new page
3) Select one of the starter page templates.
4) Select "Apply"
5) Edit the page if desired 
6) Then save or publish and exit

**Expect** 
- The event `editor_session_end` should have an additional property of `template` with the name of your selected template
- A new event titled `editor_session_template_preview` should be generated with a property of `template` with the name of your selected template
- A new event titled `editor_session_template_apply` should be generated with a property of `template` with the name of your selected template

### Start a page stop and continue
1) Open the WordPress iOS app 
2) Navigate to pages and create a new page
3) Add a title to the page
4) Save as draft 
5) Open the page again
6) Select a template and Apply it
7) Then save or publish and exit

**Expect** 
- The event `editor_session_end` should have an additional property of `template` with the name of your selected template
- A new event titled `editor_session_template_preview` should be generated with a property of `template` with the name of your selected template
- A new event titled `editor_session_template_apply` should be generated with a property of `template` with the name of your selected template

### Create a new page and cancel the template
1) Open the WordPress iOS app 
2) Navigate to pages and create a new page
3) Select one of the start page templates.
4) Select "Close"
5) Edit the page if desired 
6) Then save or publish and exit

**Expect** 
- The event `editor_session_end` should **_NOT_** have an additional property of `template` with the name of your selected template
- A new event titled `editor_session_template_preview` should be generated with a property of `template` with the name of your selected template
- `editor_session_template_apply` should **_NOT_** be generated 

## Types of changes

- Adds Analytics 

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

**Comment by pinarol** [comment]
thanks @chipsnyder looking good at the first glance 🎉 any reason we keep this draft?

**Comment by chipsnyder** [comment]
Thanks! I only left it in draft mode because we didn't have the Android side prepped yet. I doubt any changes would be needed at this layer for Android though but wanted to be sure. 

Also, I imagine Android would have issues if this is merged before that's implemented.

File: packages/block-editor/src/components/page-template-picker/picker.js [hunk 1]
```diff
@@ -25,6 +30,9 @@ const __experimentalPageTemplatePicker = ( {
 			blocks: parse( templatePreview.content ),
 		} );
 		setTemplatePreview( undefined );
+		logUserEvent( userEvents.editorSessionTemplateApply, {
+			template: templatePreview.name,
```
Comment (koke) [inline packages/block-editor/src/components/page-template-picker/picker.js]: I don't know why but I thought we had a template "slug" already. The template name can be localized, so it'd be good to add some non-translated string that uniquely identifies each template and send that instead.

Comment (chipsnyder) [reply packages/block-editor/src/components/page-template-picker/picker.js]: I think the `templatePreview` here might be the slug you're thinking of. It's just labeled differently. https://github.com/WordPress/gutenberg/pull/20317/files/1adc58f3836e3a414c231aba42eb4b2fa51d3606#diff-5ac945ef99119bbaeffb91e006ced87bR53 

I could add a `key` field to the default templates and we could use that for non translated logging.

Comment (chipsnyder) [reply packages/block-editor/src/components/page-template-picker/picker.js]: That's adjusted now

**Review by koke — APPROVED** [review]
This part also seems ready 🎉

**Pull request merged**